src/contracts/AbstractARM.sol

```diff
@@ -0,0 +1,652 @@
+        // Is there enough liquidity to claim this request?
+        require(
+            request.queued <= withdrawsClaimed + IERC20(liquidityAsset).balanceOf(address(this)),
+            "Queue pending liquidity"
+        );
+        require(request.withdrawer == msg.sender, "Not requester");
+        require(request.claimed == false, "Already claimed");
+
+        // Store the request as claimed
+        withdrawalRequests[requestId].claimed = true;
+        // Store the updated claimed amount
+        withdrawsClaimed += request.assets;
+
+        assets = request.assets;
+
+        emit RedeemClaimed(msg.sender, requestId, assets);
+
+        // transfer the liquidity asset to the withdrawer
+        IERC20(liquidityAsset).transfer(msg.sender, assets);
+    }
+
+    /// @notice Check if a withdrawal request can be claimed
+    /// @param requestId The index of the withdrawal request
+    function isClaimable(uint256 requestId) public view returns (bool) {
+        return
+            withdrawalRequests[requestId].queued <= withdrawsClaimed + IERC20(liquidityAsset).balanceOf(address(this));
+    }
+
+    /// @dev Calculate how much of the liquidity asset (WETH) in the ARM is not reserved for the withdrawal queue.
+    // That is, the amount of liquidity assets (WETH) that is available to be swapped.
+    // If there are no outstanding withdrawals, then return the maximum uint256 value.
+    // The ARM can swap out liquidity assets (WETH) that has been accrued from the performance fee for the fee collector.
+    // There is no liquidity guarantee for the fee collector. If there is not enough liquidity assets (WETH) in
+    // the ARM to collect the accrued fees, then the fee collector will have to wait until there is enough liquidity assets.
+    function _liquidityAvailable() internal view returns (uint256) {
+        // The amount of WETH that is still to be claimed in the withdrawal queue
+        uint256 outstandingWithdrawals = withdrawsQueued - withdrawsClaimed;
+
+        // Save gas on an external balanceOf call if there are no outstanding withdrawals
+        if (outstandingWithdrawals == 0) return type(uint256).max;
+
+        // The amount of the liquidity asset is in the ARM
+        uint256 liquidityBalance = IERC20(liquidityAsset).balanceOf(address(this));
+
+        // If there is not enough liquidity assets in the ARM to cover the outstanding withdrawals
+        if (liquidityBalance <= outstandingWithdrawals) {
+            return 0;
+        }
+
+        return liquidityBalance - outstandingWithdrawals;
+    }
+
+    /// @notice The total amount of assets in the ARM and external withdrawal queue,
+    /// less the liquidity assets reserved for the ARM's withdrawal queue and accrued fees.
+    function totalAssets() public view virtual returns (uint256) {
+        (uint256 fees, uint256 newAvailableAssets) = _feesAccrued();
+
+        if (fees > newAvailableAssets) return 0;
```

Comment:
🔴 I don't think we should ever be retuning a 0 from this method. We throw away some supply on launch, and it's backed by some assets. If we are ever returning low numbers here, something is not good and we get into spooky territory on our shares calculations. 

Can we just change this to a require that we are returning something above MIN_TOTAL_SUPPLY?

Reply:
Do we want to be reverting a view function?
Won't that create complications for anyone integrating? The EIP-4626 specification for `totalAssets` states "MUST NOT revert."

What if we returned `MIN_TOTAL_SUPPLY` instead of zero?

Reply:
I think the current behavior is correct.